types/check-types/index.d.ts

```diff
@@ -2,159 +2,194 @@
 // Project: https://gitlab.com/philbooth/check-types.js
 // Definitions by: idchlife <https://github.com/idchlife>
 // Definitions: https://github.com/DefinitelyTyped/DefinitelyTyped/
-// TypeScript Version: 2.2
+// Minimum TypeScript Version: 4.0
```

Comment:
Is there reason you are bumping min version supported?

Reply:
Thanks for taking a look! 

Yes, I bumped it in order to use tuples to define rest parameters at the start of the function arguments (which is how the `assert` works - it's an object with all the predicate functions modified to take an optional `string` as their last argument).

If you feel strongly that this should be avoided, the only way I can see is to explicitly define the entire assert type, which feels clumsy because it would be redefining all of the predicate functions (but with one extra argument). Of course, there might be a way that I haven't thought of.

Reply:
[lines 78-83](https://github.com/DefinitelyTyped/DefinitelyTyped/pull/51375/files#diff-dc9a0f33a5798dada648e7fc0886cd3d0550b0e7c694bd6a6df590f2b81e1474R78) are the mechanism that needs TS 4.0. Happy to modify if there's a better way to do it.